File: source/code/csharp/authentication/create_directory_linkedin.cs
using Stormpath.SDK.Provider;

var linkedInDirectory = client.Instantiate<IDirectory>();
linkedInDirectory.SetName("My LinkedIn Directory");

await client.CreateDirectoryAsync(linkedInDirectory,
    options => options.ForProvider(
        client.Providers()
            .LinkedIn()
            .Builder()
            .SetClientId("YOUR_LINKEDIN_APP_ID")
            .SetClientSecret("YOUR_LINKEDIN_APP_SECRET")
            .Build()
        ));

File: source/code/java/authentication/create_directory_linkedin.java
import com.stormpath.sdk.provider;
import com.stormpath.sdk.directory;
import com.stormpath.sdk.tenant;

Directory directory = client.instantiate(Directory.class);
directory.setName("My LinkedIn Directory");
directory.setDescription("A Github directory");

CreateDirectoryRequest request = Directories.newCreateRequestFor(directory).
            forProvider(Providers.LINKEDIN.builder()
                    .setClientId("YOUR_LINKEDIN_APP_ID")
                    .setClientSecret("YOUR_LINKEDIN_APP_SECRET")
                    .build()
            ).build();

Tenant tenant = client.getCurrentTenant();
directory = tenant.createDirectory(request);
